### PR TITLE
Support NPN fallback for Android.

### PR DESCRIPTION
**Pull request by madongfly**

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/internal/OkHttpProtocolNegotiator.java [hunk 1]
```diff
@@ -112,13 +114,69 @@ public String getSelectedProtocol(SSLSocket socket) {
...
+      // Decide which TLS Extension (APLN and NPN) we will use, follow the rules:
+      // 1. If Google Play Services Security Provider is installed, use both
+      // 2. If on Android 5.0 or later, use both, else
+      // 3. If on Android 4.1 or later, use NPN, else
+      // 4. Fail.
+      // TODO(madongfly): Logging.
+
+      // Check if Google Play Services Security Provider is installed.
+      try {
+        Class<?> securityClass = Class.forName("java.security.Security");
```
Comment (ejona86) [inline okhttp/src/main/java/com/squareup/okhttp/internal/OkHttpProtocolNegotiator.java]: Isn't `java.security.Security` always available?

Comment (madongfly) [inline okhttp/src/main/java/com/squareup/okhttp/internal/OkHttpProtocolNegotiator.java]: yes, it should be always available, I wrongly assumed the Security is a gms core class and wrote such code, then when I copied it from some import statement, I must be distracted to not even notice it is a Java class. -_-b

Thanks for the catch.

**Comment by madongfly** [comment]
@ejona86  PTAL thanks!

**Comment by ejona86** [comment]
@madongfly LGTM

**Comment by madongfly** [comment]
Manually tested 2 clients talk to a server that only supports NPN.
1. Android 5.0.1, so both ALPN and NPN are configured.
2, Android 4.4.2, so only NPN is configured.

Will check in the code soon.

**Comment by ejona86** [comment]
Thanks @madongfly!

**Pull request merged**